detox/src/devices/runtime/RuntimeDevice.js

```diff
@@ -189,9 +192,13 @@ class RuntimeDevice {
   }
 
   async installApp(binaryPath, testBinaryPath) {
-    await traceCall('appInstall', () => {
+    await traceCall('appInstall', async () => {
+      /** @type {*} */
       const currentApp = binaryPath ? { binaryPath, testBinaryPath } : this._getCurrentApp();
-      return this.deviceDriver.installApp(currentApp.binaryPath, currentApp.testBinaryPath);
+      await this.deviceDriver.installApp(currentApp.binaryPath, currentApp.testBinaryPath);
+      if (!_.isEmpty(currentApp.permissions)) {
+        await this.deviceDriver.setPermissions(currentApp.bundleId, currentApp.permissions);
+      }
```

Comment:
Well I didn't go deeply into this but something about this approach feels a bit off. It assumes that the permissions can be set in such a global way - before an instance of the app is running. While it works on iOS, I can't 100% say the same about Android / a theoretical future platform.

What about applying a similar solution to the launch-args technique, which is set in advance and handled on-site (i.e. in launch app)?